### ISSUE TITLE
moto mocking not working. 

### ISSUE THREAD

**Issue by asahasrabuddhe** [spulec/moto#1815]
Hello,

I am trying to write tests for a serverless application. I am using serverless framework. I am facing a weird issue. Whenever I try to mock s3 or dynamodb using moto, it does not work. Instead of mocking, the boto3 call actually goes to my aws account and tries to do things there.

This is not desirable behaviour. Could you please help?

**Comment by asahasrabuddhe** [comment]
Sample Code :

`class TestJobEngine(TestCase):
    @mock.patch.dict(os.environ, {'IN_QUEUE_URL': 'mytemp'})
    @mock.patch('JobEngine.job_engine.logger')
    @mock_dynamodb2
    def test_check_duplicate(self, mock_logger):
        job_id = 'ABCD123'
        url = 'http://www.google.com'
        status_code = '0'
        html_status_code = '0'
        screenshot_url = 'sad'
        max_depth = '3'
        max_pages = '6'
        scrape_type = '1'
        proxy = 's'

        db = boto3.resource('dynamodb', 'us-east-1')

        table = db.create_table(
            TableName='jobs',
            KeySchema=[
                {
                    'AttributeName': 'id',
                    'KeyType': 'HASH'
                }
            ],
            AttributeDefinitions=[
                {
                    'AttributeName': 'id',
                    'AttributeType': 'S'
                }
            ],
            ProvisionedThroughput={
                'ReadCapacityUnits': 1,
                'WriteCapacityUnits': 1
            }
        )

        table.meta.client.get_waiter('table_exists').wait(TableName='jobs')

        table.put_item(
            Item={
                'id': {'S': str(uuid.uuid5(uuid.NAMESPACE_DNS, job_id))},
                'job_id': {'S': job_id},
                'url': {'S': url},
                'status_code': {'S': status_code},
                'html_status_code': {'S': html_status_code},
                'screenshot_url': {'S': screenshot_url},
                'max_depth': {'S': max_depth},
                'max_pages': {'S': max_pages},
                'scrape_type': {'S': scrape_type},
                'proxy': {'S': proxy},
                'created_at': {'S': datetime.datetime.utcnow().isoformat()}
            }
        )

        res = check_duplicate(job_id)
        self.assertTrue(mock_logger.info.called)
        self.assertEqual(res, True, 'True')`

**Comment by asahasrabuddhe** [comment]
```
import datetime
import boto3
import uuid
import os
from moto import mock_dynamodb2
from unittest import mock, TestCase
from JobEngine.job_engine import check_duplicate


class TestJobEngine(TestCase):
    @mock.patch.dict(os.environ, {'IN_QUEUE_URL': 'mytemp'})
    @mock.patch('JobEngine.job_engine.logger')
    @mock_dynamodb2
    def test_check_duplicate(self, mock_logger):
        job_id = 'ABCD123'
        url = 'http://www.google.com'
        status_code = '0'
        html_status_code = '0'
        screenshot_url = 'sad'
        max_depth = '3'
        max_pages = '6'
        scrape_type = '1'
        proxy = 's'

        db = boto3.resource('dynamodb', 'us-east-1')

        table = db.create_table(
            TableName='jobs',
            KeySchema=[
                {
                    'AttributeName': 'id',
                    'KeyType': 'HASH'
                }
            ],
            AttributeDefinitions=[
                {
                    'AttributeName': 'id',
                    'AttributeType': 'S'
                }
            ],
            ProvisionedThroughput={
                'ReadCapacityUnits': 1,
                'WriteCapacityUnits': 1
            }
        )

        table.meta.client.get_waiter('table_exists').wait(TableName='jobs')

        table.put_item(
            Item={
                'id': {'S': str(uuid.uuid5(uuid.NAMESPACE_DNS, job_id))},
                'job_id': {'S': job_id},
                'url': {'S': url},
                'status_code': {'S': status_code},
                'html_status_code': {'S': html_status_code},
                'screenshot_url': {'S': screenshot_url},
                'max_depth': {'S': max_depth},
                'max_pages': {'S': max_pages},
                'scrape_type': {'S': scrape_type},
                'proxy': {'S': proxy},
                'created_at': {'S': datetime.datetime.utcnow().isoformat()}
            }
        )

        res = check_duplicate(job_id)
        self.assertTrue(mock_logger.info.called)
        self.assertEqual(res, True, 'True')
```

**Comment by asahasrabuddhe** [comment]
Please see the above code, I am trying to insert a record into the table and then call a function that would verify if the specified id is already present in the table. Here, I get an error table already exists when I run this code. 

If I disable the network, I get an error ```botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "https://dynamodb.us-east-1.amazonaws.com/"
```

I fail to understand why there is an attempt to connect to aws if we are trying to mock.

**Comment by asahasrabuddhe** [comment]
Hi,

The issue was due to a newer issue of botocore. See https://github.com/spulec/moto/issues/1793

This was fixed by downgrading botocore to 1.10.84

Thanks!
